sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -1545,6 +1545,59 @@ class DataFrameAggregateSuite extends QueryTest
     )
     checkAnswer(res, Row(Array(1), Array(1)))
   }
+
+  test("SPARK-16484: hllsketch_estimate positive tests") {
```

Comment:
we are going to need more test cases :) some ideas:

* empty input table
* use SQL to query the aggregate functions
* explicit regular integers vs. long integers in the input table
* minimum and maximum supported values for each data type

Reply:
Some more testing ideas:
* Table that contains only `NULL` values.
* Table that contains `NULL` and non-`NULL` values.
* Table with two columns where the first column is used for grouping (grouping aggregation) and the second is the one we want to compute the sketch for. In that table, I would like to see one group that has only `NULL` values, and one group that has both `NULL` and non-`NULL` values. So we are talking about at least 2 different groups.

Reply:
Resolving this comment as I've added tests for all ideas listed, except for the `Merging sparse sketches with dense sketches` idea which I think we should rely on the datasketches tests to cover.